syntax = "proto3";

service StockService {
    rpc batch_init_users(BatchInitRequest) returns (BatchInitResponse);
    rpc create_item(CreateItemRequest) returns (CreateItemResponse);
    rpc find_item(FindItemRequest) returns (FindItemResponse);
    rpc add_stock(StockRequest) returns (StockResponse);
    rpc remove_stock(StockRequest) returns (StockResponse);
}

message BatchInitRequest {
    int32 n = 1;
    int32 starting_money = 2;
    int32 item_price = 3;
}

message BatchInitResponse {
    string message = 1;
}

message CreateItemRequest {
    int32 price = 1;
}

message CreateItemResponse {
    string item_id = 1;
}

message FindItemRequest {
    string item_id = 1;
}

message FindItemResponse {
    int32 stock = 1;
    int32 price = 2;
}

message StockRequest {
    string item_id = 1;
    int32 amount = 2;
}

message StockResponse {
    int32 statuscode = 1;
}